scripts/benchmarks/convert_pytest_JSON_to_XUBM.py

```diff
@@ -0,0 +1,104 @@
+        help="Name of the XUBM JSON file to be exported.",
+    )
+
+    parser.add_argument(
+        "--author",
+        type=str,
+        nargs="?",
+        help="User associated with the commit.",
+    )
+
+    parser.add_argument(
+        "--github_reference",
+        type=str,
+        default="",
+        nargs="?",
+        help="The fully-formed ref of the branch or tag that triggered the workflow run.",
+    )
+
+    return parser.parse_args()
+
+def create_benchmark_XUBM(stored_data, data, args):
+    """This function converts the JSON file provided by pytest-benchmark to the JSON-XUBM (XanadU BenchMark) format.
+
+    Args:
+        stored_data: data we already have in-place or an empty dictionary.
+        data: Pytest-benchmark JSON data
+        args: customization arguments
+
+    Returns:
+        JSON-XUBM: JSON data
+    """
+    commit_reference = args.github_reference if (args.github_reference != "") else data["commit_info"]["branch"]
+    if commit_reference not in stored_data:
+        stored_data[commit_reference] = {}
+    else:
+        print("data already exist. doing nothing.")
+        return stored_data
+
+    hash_commit = hash(data["commit_info"]["time"]) + hash(commit_reference)
+    for benchmark in data["benchmarks"]:
+        benchmark_xubm = {}
+        benchmark_xubm["uid"] = hash_commit + hash(benchmark["fullname"]) + hash(json.dumps(benchmark["params"]))
+        benchmark_xubm["date"] = data["commit_info"]["time"]
+        benchmark_xubm["fullname"] = benchmark["fullname"]
+        benchmark_xubm["gitID"] = data["commit_info"]["id"]
+        benchmark_xubm["runs"] = benchmark["stats"]["iterations"]
+        benchmark_xubm["params"] = benchmark["params"]
+        benchmark_xubm["runtime"] = benchmark["stats"]["mean"]
+        benchmark_xubm["timeUnit"] = "seconds"
+        benchmark_xubm["project"] = data["commit_info"]["project"]
+        benchmark_xubm["machine"] = data["machine_info"]
+        benchmark_xubm["users"] = args.author
+        # Pytest-benchmark provides a reach variety of statistics. This will be stored as metadata.
+        benchmark_xubm["metadata"] = {"stats": benchmark["stats"]}
+
+        stored_data[commit_reference][benchmark["name"]] = benchmark_xubm
+    return stored_data
+
+if __name__ == "__main__":
+    parsed_args = parse_args()
+
+    if os.stat(parsed_args.filename).st_size == 0:
+        print(parsed_args.filename+" is empty. Interrupting program.")
+        sys.exit(0)
+
+    with open(parsed_args.filename, 'r', encoding="utf-8") as file:
+        pytest_data = json.load(file)
+
+    # Check if the JSON-XUBM file already exist, or if we'll start with and empty dictionary.
+    if os.path.isfile(parsed_args.filename_XUBM):
+        with open(parsed_args.filename_XUBM, 'r', encoding="utf-8") as file:
+            xubm_database = json.load(file)
+    else:
+        xubm_database = {}
+
+    XUBM_data = create_benchmark_XUBM(xubm_database, pytest_data, parsed_args)
```

Comment:
the script is pretty private and isolated but I would prefer to avoid editing dictionaries passed as args to another function in-place unless totally necessary

Reply:
Hi. Here I'm initializing the dictionary, with data or empty depending on the condition of having such data. I don't see it as editing the dictionary in place. Could you please expand on that? Do you have a suggested alternative?

Reply:
the in-place editing is that after line 101 here, `xubm_database` has been changed which is unexpected (it is the same reference as `XUBM_data` now). That said, you don't refer to `xubm_database` again so it shouldn't matter, so don't worry about it 😄 I see you did it to compute the commit reference within `create_benchmark_XUBM` and isolate all logic to there. looks good 👍